packages/in_app_purchase/in_app_purchase_storekit/darwin/Classes/InAppPurchasePlugin.swift

```diff
@@ -0,0 +1,412 @@
+      }
+    #endif
+  }
+
+  public func removePaymentQueueDelegateWithError(
+    _ error: AutoreleasingUnsafeMutablePointer<FlutterError?>
+  ) {
+    #if os(iOS)
+      if #available(iOS 13.0, *) {
+        paymentQueueDelegateCallbackChannel = nil
+        paymentQueueHandler!.delegate = nil
+        paymentQueueDelegate = nil
+      }
+    #endif
+  }
+
+  public func showPriceConsentIfNeededWithError(
+    _ error: AutoreleasingUnsafeMutablePointer<FlutterError?>
+  ) {
+    #if os(iOS)
+      if #available(iOS 13.4, *) {
+        paymentQueueHandler!.showPriceConsentIfNeeded()
+      }
+    #endif
+  }
+
+  public func handleTransactionsUpdated(_ transactions: [SKPaymentTransaction]) {
+    var maps: [[AnyHashable: Any]] = []
+    for transaction in transactions {
+      let map = FIAObjectTranslator.getMapFrom(transaction)
+      maps.append(map)
+    }
+    transactionObserverCallbackChannel!.invokeMethod("updatedTransactions", arguments: maps)
+  }
+
+  public func handleTransactionsRemoved(_ transactions: [SKPaymentTransaction]) {
+    var maps: [[AnyHashable: Any]] = []
+    for transaction in transactions {
+      let map = FIAObjectTranslator.getMapFrom(transaction)
+      maps.append(map)
+    }
+    transactionObserverCallbackChannel!.invokeMethod("removedTransactions", arguments: maps)
+  }
+
+  public func handleTransactionRestoreFailed(_ error: NSError) {
+    transactionObserverCallbackChannel!.invokeMethod(
+      "restoreCompletedTransactionsFailed", arguments: FIAObjectTranslator.getMapFrom(error))
+  }
+
+  public func restoreCompletedTransactionsFinished() {
+    transactionObserverCallbackChannel!.invokeMethod(
+      "paymentQueueRestoreCompletedTransactionsFinished", arguments: nil)
+  }
+
+  public func shouldAddStorePayment(payment: SKPayment, product: SKProduct) -> Bool {
+    productsCache[product.productIdentifier] = product
+    transactionObserverCallbackChannel!
+      .invokeMethod(
+        "shouldAddStorePayment",
+        arguments: [
+          "payment": FIAObjectTranslator.getMapFrom(payment),
+          "product": FIAObjectTranslator.getMapFrom(product),
+        ])
+    return false
+  }
+
+  func updatedDownloads() {
+    NSLog("Received an updatedDownloads callback, but downloads are not supported.")
+  }
+
+  public func canMakePayments() -> Bool {
+    return SKPaymentQueue.canMakePayments() as Bool
+  }
+
+  func getProduct(productID: String) -> SKProduct? {
+    return self.productsCache[productID] as? SKProduct
+  }
+
+  func getProductRequest(withIdentifiers productIdentifiers: Set<String>) -> SKProductsRequest {
+    return SKProductsRequest(productIdentifiers: productIdentifiers)
+  }
+
+  func getNonNullValue(from dictionary: [String: Any], forKey key: String) -> Any? {
+    let value = dictionary[key]
+    return value is NSNull ? nil : value
+  }
+
+  func getRefreshReceiptRequest(properties: [String: Any]?) -> SKReceiptRefreshRequest {
```

Comment:
are these helpers only used inside this file? should use `private` if that's the case.

Reply:
this

Reply:
these need to be not private so they can be overriden for testing purposes
Like here:
https://github.com/flutter/packages/blob/74de10af17274ab3d93a264b215ba19ec20e7ba6/packages/in_app_purchase/in_app_purchase_storekit/example/ios/RunnerTests/SwiftStubs.swift#L11